v2/protocol/http/protocol.go

```diff
@@ -262,6 +262,7 @@ func (p *Protocol) ServeHTTP(rw http.ResponseWriter, req *http.Request) {
 	m := NewMessageFromHttpRequest(req)
 	if m == nil || m.ReadEncoding() == binding.EncodingUnknown {
 		p.incoming <- msgErr{msg: nil, err: binding.ErrUnknownEncoding}
+		rw.WriteHeader(http.StatusUnsupportedMediaType)
```

Comment:
Nope :smile:

Reply:
This breaks this use case: https://github.com/cloudevents/sdk-go/blob/master/v2/cmd/samples/message-handle-non-cloudevents/main.go

http.UnsupportedMediaType is returned by that https://github.com/cloudevents/sdk-go/blob/master/v2/protocol/http/protocol.go#L259

Reply:
I tried to find where the spec says this and was unable to find any data. It seems real wrong to send back a 200 if it is not a cloudevent. The only way to get to this code is if it is expecting to get a cloudevent.

Reply:
But that means that we are impeding to use the sdk to receive non events completely? If that's the case, then we need to proper document it and make this behaviour consistent between protocols

Reply:
you can still map a path to a custom handler, there are a few examples of that. But for events that are directed to the sdk handler, and the request is not a CloudEvent, it should be an error.

Reply:
What I'm saying is that i don't see how "leaving this door open" could be a problem

Reply:
in this codepath, there is no chance to get to the users code of the sdk, it is returning `HTTP - 200 OK` when the return hits. I think that is a bug because it means the caller can do things like post a blank message to the sdk handler and it will get back `200 OK`. We need to tell the caller they done screwed up, or are calling an end point that has no idea what you are talking about. In this codepath the sdk bails out. It is totally an error on the callers side and not `OK`.

Reply:
Note that `msg` is `nil`. which means the request is not held open like in other cases.

Reply:
ok removed.